src/NuGet.Core/NuGet.Packaging/RuntimeModel/RuntimeGraph.cs

```diff
@@ -8,98 +8,164 @@
 using System.Linq;
 using NuGet.Shared;
 
+#nullable enable
+
 namespace NuGet.RuntimeModel
 {
-    public class RuntimeGraph : IEquatable<RuntimeGraph>
+    public sealed class RuntimeGraph : IEquatable<RuntimeGraph>
     {
-        private readonly ConcurrentDictionary<RuntimeCompatKey, bool> _areCompatible
-            = new ConcurrentDictionary<RuntimeCompatKey, bool>();
-
-        private readonly ConcurrentDictionary<string, HashSet<string>> _expandCache
-            = new ConcurrentDictionary<string, HashSet<string>>(StringComparer.Ordinal);
+        private static readonly ReadOnlyDictionary<string, RuntimeDescription> EmptyRuntimes = new(new Dictionary<string, RuntimeDescription>());
+        private static readonly ReadOnlyDictionary<string, CompatibilityProfile> EmptySupports = new(new Dictionary<string, CompatibilityProfile>());
```

Comment:
These singleton empty dictionaries are used in constructor chaining (to a new constructor) instead of passing `Enumerable.Empty<>` to the prior constructor which would then construct a new empty dictionary. Actually, it would do that twice, as it then wrapped it in `ReadOnlyDictionary<>`.